grails-app/services/com/unifina/service/StreamService.groovy

```diff
@@ -116,8 +116,7 @@ class StreamService {
 	}
 
 	// Ref to Kafka will be abstracted out when Feed abstraction is reworked
-	@CompileStatic
-	void sendMessage(Stream stream, @Nullable String partitionKey, long timestamp, byte[] content, byte contentType, int ttl=0) {
+	void sendMessage(Stream stream, String partitionKey, long timestamp, byte[] content, byte contentType, int ttl=0) {
```

Comment:
@harbu Was it the `@CompileStatic` that was causing the crash? This is one of the most frequently called methods, would be nice to have that performance benefit there. If it's a no-go, then.. no can do.

Reply:
Not necessarily caused by `@CompileStatic`, but probably due to overloading of method name.

I can add the `@CompileStatic` back in and run all tests when I have a chance.

Reply:
I've tried it now and its looking good!

Reply:
First I ran all tests once without the CompileStatic annotation patch and no SIGSEGV.
Second I've run all tests twice with the CompileStatic annotation patch above and they passed without SIGSEGV!